IBAnimatable/PresentationDesignable.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Created by Tom Baranes on 16/07/16.
+//  Copyright © 2016 Jake Lin. All rights reserved.
+//
+
+import UIKit
+
+public protocol PresentationDesignable: class {
+
+  var presentationAnimationType: String? { get set }
+  var dismissAnimationType: String? { get set }
+  var transitionDuration: Double { get set }
+  var modalPosition: String { get set }
+  var modalWidth: String { get set }
+  var modalHeight: String { get set }
+
+  var cornerRadius: CGFloat { get set }
+  var dismissOnTap: Bool { get set }
+
+  var backgroundColor: UIColor { get set }
+  var opacity: CGFloat { get set }
+
+  var blurEffectStyle: String? { get set }
+  var blurOpacity: CGFloat { get set }
+
+  var shadowColor: UIColor? { get set }
+  var shadowRadius: CGFloat { get set }
+  var shadowOpacity: CGFloat { get set }
+  var shadowOffset: CGPoint { get set }
+}
+
+public struct PresentedSetup {
```

Comment:
I really like this struct. I am thinking the name of it. `Setup` is also a noun, but I feel `configuration` maybe better for a struct. How about `PresentionConfiguration`? It's very challenging for me to pick up a good name, 😂